test/embedded/init-throwing.swift

```diff
@@ -0,0 +1,66 @@
+// RUN: %target-run-simple-swift(-enable-experimental-feature Embedded -runtime-compatibility-version none -wmo -Xfrontend -disable-objc-interop) | %FileCheck %s
+
+// REQUIRES: swift_in_compiler
+// REQUIRES: executable_test
+// REQUIRES: optimized_stdlib
+// REQUIRES: OS=macosx || OS=linux-gnu
+
+public struct FooError: Error {}
+
+public class PrintingClass {
+    init() { print("PrintingClass.init") }
+    deinit { print("PrintingClass.deinit") }
+}
+
+public class Foo {
+    var a: PrintingClass
+    var b: PrintingClass
+    init(shouldThrow: Bool) throws(FooError) {
+        a = PrintingClass()
+        if shouldThrow { throw FooError() }
+        b = PrintingClass()
+    }
+}
```

Comment:
To really exercise the new code path, you'll need some subclasses of `Foo`. For example, I recommend one with trivially-destructible storage and one with non-trivially-destructible storage, e.g.,

```swift
public class Bar: Foo {
  var value: Int = 17
}

public class Wibble: Bar {
  var c: PrintingClass = .init()
}
```

When initializing a `Wibble(shouldThrow: true)`, we'll go through the ivar destroyers for the `Wibble` and `Bar` instances. (The `Bar` one should be NULL, per my comments earlier).

Reply:
I think this is covered below in this file? Or is a 3-subclass hierarchy also an interesting case to add?

Reply:
You don't need a 3-subclass hierarchy per se, (although it's nice to go through the loop more than once), but you do need a case where none of the stored properties need to be destroyed.

Reply:
Added the testcase, thanks!